src/test/java/liuni/TwitterServiceTest.java

```diff
@@ -0,0 +1,204 @@
+        twitterService = TwitterService.getInstance();
+        twitter = mock(Twitter.class);
+        twitterService.setTwitter(twitter);
+    }
+
+    private void setUpUserBadURL() {
+        testUserName = "TestUserName";
+        testUserScreenName = "TestUserScreenName";
+        testUserProfileURL = "BadURL";
+
+        status = mock(Status.class);
+        User user = mock(User.class);
+        when(user.getName()).thenReturn(testUserName);
+        when(user.getScreenName()).thenReturn(testUserScreenName);
+        when(user.getProfileImageURL()).thenReturn(testUserProfileURL);
+        when(status.getUser()).thenReturn(user);
+    }
+
+    private void setUpUserGoodURL() {
+        testUserName = "TestUserName";
+        testUserScreenName = "TestUserScreenName";
+        testUserProfileURL = "http://abs.twimg.com/sticky/default_profile_images/default_profile_normal.png";
+
+        testURL = null;
+        try {
+            testURL = new URL(testUserProfileURL);
+        }
+        catch (Exception e) {
+            Assert.fail("This exception is not expected.");
+        }
+
+        assertTrue(testURL != null);
+
+        status = mock(Status.class);
+        User user = mock(User.class);
+        when(user.getName()).thenReturn(testUserName);
+        when(user.getScreenName()).thenReturn(testUserScreenName);
+        when(user.getProfileImageURL()).thenReturn(testUserProfileURL);
+        when(status.getUser()).thenReturn(user);
+    }
+
+    @Test
+    public void testGetUserBadURL() {
+        setUpUserBadURL();
+        UserModel resultModel = twitterService.getUser(status);
+        assertEquals(testUserName, resultModel.getName());
+        assertEquals(testUserScreenName, resultModel.getTwitterHandle());
+        assertEquals(null, resultModel.getProfileImageUrl());
+    }
+
+    @Test
+    public void testGetUserGoodURL() {
+        setUpUserGoodURL();
+        UserModel resultModel = twitterService.getUser(status);
+        assertEquals(testUserName, resultModel.getName());
+        assertEquals(testUserScreenName, resultModel.getTwitterHandle());
+        assertEquals(testURL, resultModel.getProfileImageUrl());
+    }
+
+    private void setUpTweet() {
+        date = new Date(2323223232L);
+        testMessage = "Test Message";
+        when(status.getText()).thenReturn(testMessage);
+        when(status.getCreatedAt()).thenReturn(date);
+    }
+
+    @Test
+    public void test_getTweet() {
+        setUpUserBadURL();
+        setUpTweet();
+
+        TwitterTweetModel tweetModel = twitterService.getTweet(status);
+        assertEquals(date, tweetModel.getCreatedAt());
+        assertEquals(testMessage, tweetModel.getMessage());
+        assertEquals(testUserName, tweetModel.getUser().getName());
+        assertEquals(testUserScreenName, tweetModel.getUser().getTwitterHandle());
+        assertEquals(null, tweetModel.getUser().getProfileImageUrl());
+    }
+
+    private TwitterTweetModel createTweetModel() {
```

Comment:
instead of two entirely separate model definitions, you could create a helper function to create objects based on data passed in

Reply:
Done.